src/bin/ion/commands/beta/generate/generator.rs

```diff
@@ -65,56 +114,46 @@ impl<'a> CodeGenerator<'a> {
         ))
     }
 
-    /// Generates code for given Ion Schema
-    pub fn generate(&mut self, schema: IslSchema) -> CodeGenResult<()> {
-        // this will be used for Rust to create mod.rs which lists all the generated modules
-        let mut modules = vec![];
-        let mut module_context = tera::Context::new();
-
-        // Register a tera filter that can be used to convert a string to upper camel case
-        self.tera.register_filter("upper_camel", Self::upper_camel);
-
-        for isl_type in schema.types() {
-            self.generate_data_model(&mut modules, isl_type)?;
-        }
-
-        if self.language == Language::Rust {
-            module_context.insert("modules", &modules);
-            let rendered = self.tera.render("rust/mod.templ", &module_context)?;
-            let mut file = File::create(self.output.join("mod.rs"))?;
-            file.write_all(rendered.as_bytes())?;
-        }
-
-        Ok(())
+    /// Represents a [tera] filter that return true if the value is a built in type, otherwise returns false.
+    ///
+    /// For more information: <https://docs.rs/tera/1.19.0/tera/struct.Tera.html#method.register_filter>
+    ///
+    /// [tera]: <https://docs.rs/tera/latest/tera/>
+    pub fn is_built_in_type(
```

Comment:
(🗺️ PR tour) This is a filter method that is sued in templates for read and write API generation. Previously the code to generate read and write API was inside `CodeGenerator` but now it has been moved to the template itself for ease of maintenance.